src/main/java/com/mucommander/ui/main/MainFrame.java

```diff
@@ -260,11 +261,16 @@ public void windowClosing(WindowEvent e) {
 
         ActionKeymap.registerActions(this);
 
-        // Fire table change events on registered ActivePanelListener instances, to notify of the intial active table.
+        // Fire table change events on registered ActivePanelListener instances, to notify of the initial active table.
         fireActivePanelChanged(activeTable.getFolderPanel());
 
-        // Set the custom FocusTraversalPolicy that manages focus for both FolderPanel and their subcomponents.
+        // Set the custom FocusTraversalPolicy that manages focus for both FolderPanel and their sub components.
         setFocusTraversalPolicy(new CustomFocusTraversalPolicy());
+
+        // Restore single panel view state
+        if(MuConfigurations.getPreferences().getVariable(MuPreference.USE_SINGLE_PANEL_VIEW, MuPreferences.DEFAULT_USE_SINGLE_PANEL_VIEW)) {
+            ActionManager.performAction(ToggleUseSinglePanelAction.Descriptor.ACTION_ID, this);
+        }
```

Comment:
Yeah, I agree with what you say and that it becomes complex :) So we can handle it separately. Let's try to extract this part to DefaultMainFramesBuilder so the configuration from snapshot.xml and preferences.xml will be applied on the main frames that are created on startup only for now.